resource-sharing-partners/src/main/java/no/sikt/rsp/clients/alma/HttpUrlConnectionAlmaPartnerUpdater.java

```diff
@@ -0,0 +1,244 @@
+
+    private static final String ALMA_ERROR_CODE_PARTNER_NOT_FOUND = "402118";
+    private static final String AUTHORIZATION_HEADER_NAME = "Authorization";
+    private static final String ACCEPT_HEADER_NAME = "Accept";
+    public static final String CONTENT_TYPE_HEADER_NAME = "Content-Type";
+    private static final String APIKEY_KEY = "apikey";
+    private static final String SPACE_KEY = " ";
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String APPLICATION_XML = "application/xml";
+    public static final String PARTNERS_URL_PATH = "partners";
+
+    private final transient String almaApikey;
+    private final transient URI almaApiHost;
+    private static final ObjectMapper objectMapper = new ObjectMapper()
+                                                         .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES,
+                                                                    false);
+
+    public HttpUrlConnectionAlmaPartnerUpdater(final String almaApiKey, final URI almaApiHost) {
+        this(HttpClient.newBuilder().version(HttpClient.Version.HTTP_2).build(), almaApiKey, almaApiHost);
+    }
+
+    public HttpUrlConnectionAlmaPartnerUpdater(final HttpClient httpClient, final String almaApiKey,
+                                               final URI almaApiHost) {
+        super(httpClient);
+        this.almaApikey = almaApiKey;
+        this.almaApiHost = almaApiHost;
+    }
+
+    private Optional<String> getPartner(final String code) {
+        final HttpRequest request = HttpRequest.newBuilder()
+                                        .GET()
+                                        .uri(UriWrapper.fromUri(almaApiHost)
+                                                 .addChild(PARTNERS_URL_PATH)
+                                                 .addChild(code).getUri())
+                                        .setHeader(AUTHORIZATION_HEADER_NAME, APIKEY_KEY + SPACE_KEY + almaApikey)
+                                        .setHeader(ACCEPT_HEADER_NAME, APPLICATION_JSON)
+                                        .build();
+        try {
+            final HttpResponse<String> response = httpClient.send(request, BodyHandlers.ofString());
+            if (response.statusCode() == HttpURLConnection.HTTP_OK) {
+                return Optional.of(response.body());
+            } else if (HttpURLConnection.HTTP_BAD_REQUEST == response.statusCode()) {
+                final AlmaErrorResponse almaErrorResponse = objectMapper.readValue(response.body(),
+                                                                                   AlmaErrorResponse.class);
+                if (partnerNotFound(almaErrorResponse)) {
+                    return Optional.empty();
+                } else {
+                    final String message = String.format(
+                        "Unexpected error response getting partner '%s' from Alma. Status code %d, body '%s'.",
```

Comment:
Jeg foretrekker å har disse som Constant. Da kan jeg refere til disse i test for eksempel.